### PR TITLE
feat(sdk)!: response.body is str

### PR DESCRIPTION
**Pull request by Chriscbr**
Currently a lot of Wing code that deals with API responses has to perform extra checking to handle `body` fields that may be `nil`. For example:

```js
let response = http.get(...);
let foo = Foo.parseJson(response.body ?? "");
log(foo.bang);
```

Notice above how we have to write `response.body ?? ""` since `response.body` is typed as `str?`.

Since the `http` module calls the [text()](https://developer.mozilla.org/en-US/docs/Web/API/Response/text) method in JavaScript which always returns a `string`, it seems as though it ought to be safe to change the type of `body` to `str` and provide a better DX in the standard library:

```js
let response = http.get(...);
let foo = Foo.parseJson(response.body); // changed
log(foo.bang);
```

Another example with a HEAD request: 
```js
test "head request" {
  let resp = http.fetch("https://news.ycombinator.com", {
    method: http.HttpMethod.HEAD,
  });
  assert(resp.body == "");
}
```

## Checklist

- [x] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [x] Description explains motivation and solution
- [x] Tests added (always)
- [ ] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

**Comment by garysassano** [comment]
I guess this is related to #4847 ?

File: examples/proposed/url-shortener.main.w [hunk 1]
```diff
@@ -11,14 +11,14 @@ class UrlShortener {
...
-  inflight getId(url: str): str {
+  pub inflight getId(url: str): str {
     let id = this._makeId();
     this.lookup.put(id, url);
     return id;
   }
 
   // Get the url for the given id. Returns nil if the id is invalid.
-  inflight getUrl(id: str): str? {
+  pub inflight getUrl(id: str): str? {
```
Comment (MarkMcCulloh) [inline examples/proposed/url-shortener.main.w:21]: Why the modifier change?

**Review by MarkMcCulloh — APPROVED** [review]
LGTM
I believe this should be a breaking change

File: examples/proposed/url-shortener.main.w (hunk 1 above)
Comment (Chriscbr) [reply examples/proposed/url-shortener.main.w:21]: Ah - this is code from the proposed examples folder. It's not getting compiled by CI but I just came around to updating the code

**Comment by Chriscbr** [comment]
> I guess this is related to #4847 ?

That issue requires a separate change since it pertains to the Function class

**Review by eladb — APPROVED** [review]

**Pull request merged**